app/fetchers/space_users_list_fetcher.rb

```diff
@@ -0,0 +1,40 @@
+require 'fetchers/base_list_fetcher'
+
+module VCAP::CloudController
+  class SpaceUsersListFetcher < BaseListFetcher
+    class << self
+      def fetch_all(message, space, readable_users_dataset)
+        filter(message, space, readable_users_dataset)
+      end
+
+      private
+
+      def filter(message, space, dataset)
+        if message.requested?(:usernames)
+          guids = uaa_client.ids_for_usernames_and_origins(message.usernames, message.origins)
+          dataset = dataset.where(guid: guids)
+        end
+
+        if message.requested?(:label_selector)
+          dataset = LabelSelectorQueryGenerator.add_selector_queries(
+            label_klass: UserLabelModel,
+            resource_dataset: dataset,
+            requirements: message.requirements,
+            resource_klass: User,
+          )
+        end
+
+        dataset = super(message, dataset, User)
+
+        dataset.inner_join(SpaceDeveloper.table_name, user_id: :id).where(space_id: space.id).select_all(:users).
```

Comment:
Does it make more sense to keep this logic at the model layer? We already have an [established pattern for visible org users](https://github.com/cloudfoundry/cloud_controller_ng/blob/2860386d3f0a8402df451a52858df321f104fd44/app/models/runtime/user.rb#L188)

Reply:
I didn't look at user.rb before.  Cool with me to move it if that's the pattern we like